script/TrainClass.lua

```diff
@@ -81,6 +81,9 @@ function TrainClass:isEnoughFuel()
         for _, loco in pairs(dir) do
             local burner = loco.burner
             if burner then
+                if #burner.inventory == 0 then
```

Comment:
I will need to test before merging into the master branch. Please do not make several different changes in one PR.

Reply:
Of course. Please let me know if I made a mistake. This is my first time working on a mod for factorio and in LUA. And also with using github/working with others. :)

This fixed the issue with the other mod from my testing, and should work for any train that works without a fuel inventory. I'm not sure how it would work with mods like Steam Trains  or Fluid Trains, as I haven't used those.

Reply:
I think this is not a universal solution and in the future with some mod it will not work. But while there is no such request, you can leave it like that.